owncloudComLibrary/src/main/java/com/owncloud/android/lib/common/http/interceptors/HttpInterceptor.java

```diff
@@ -46,26 +46,32 @@
     public Response intercept(Chain chain) throws IOException {
         Request request = chain.request();
 
-        for (RequestInterceptor interceptor : mRequestInterceptors) {
-            request = interceptor.intercept(request);
+        ListIterator<RequestInterceptor> requestInterceptorIterator = mRequestInterceptors.listIterator();
+
+        while (requestInterceptorIterator.hasNext()) {
+            RequestInterceptor currentRequestInterceptor = requestInterceptorIterator.next();
+            request = currentRequestInterceptor.intercept(request);
         }
 
         Response response = chain.proceed(request);
 
-        for (ResponseInterceptor interceptor : mResponseInterceptors) {
-            response = interceptor.intercept(response);
+        ListIterator<ResponseInterceptor> responseInterceptorIterator = mResponseInterceptors.listIterator();
+
+        while (responseInterceptorIterator.hasNext()) {
```

Comment:
I think you would change this code. Let me guess, In a coming PR??

Reply:
Yes